api/src/pcapi/routes/backoffice_v3/collective_offers/blueprint.py

```diff
@@ -318,14 +322,116 @@ def batch_reject_collective_offers() -> utils.BackofficeResponse:
         flash("L'un des identifiants sélectionnés est invalide", "danger")
         return redirect(request.referrer or url_for("backoffice_v3_web.collective_offer.list_collective_offers"), 303)
 
-    _batch_validate_or_reject_collective_offers(OfferValidationStatus.REJECTED, form.object_ids_list)
+    _batch_validate_or_reject_collective_offers(offer_mixin.OfferValidationStatus.REJECTED, form.object_ids_list)
     return redirect(request.referrer or url_for("backoffice_v3_web.collective_offer.list_collective_offers"), 303)
 
 
-@list_collective_offers_blueprint.route("/<int:collective_offer_id>/details", methods=["GET"])
+def _is_collective_offer_price_editable(collective_offer: educational_models.CollectiveOffer) -> bool:
+    # if the offer has no stock it has no price therefore its price cannot be updated
+    if not collective_offer.collectiveStock:
+        return False
+
+    # cannot update an offer's price while the cashflow generation script is running
+    if app.redis_client.exists(finance_conf.REDIS_GENERATE_CASHFLOW_LOCK):  # type: ignore [attr-defined]
+        return False
+
+    # cannot update an offer's stock if it is already a processed pricing
+    pricing_query = db.session.query(finance_models.Pricing.id)
+    pricing_query = pricing_query.join(educational_models.CollectiveBooking, finance_models.Pricing.collectiveBooking)
+    pricing_query = pricing_query.join(
+        educational_models.CollectiveStock, educational_models.CollectiveBooking.collectiveStock
+    )
+    pricing_query = pricing_query.filter(
+        educational_models.CollectiveStock.collectiveOfferId == collective_offer.id,
+        finance_models.Pricing.status.in_(
+            (finance_models.PricingStatus.PROCESSED, finance_models.PricingStatus.INVOICED)
+        ),
+    )
```

Comment:
On peut pas faire un pricing_query = gnagna.query.join(...).join(...).filter(...) ? C'est plus simple à lire

Reply:
c'est marrant, je trouve ta proposition plus difficile a lire pour le coup.

Reply:
Pour mettre mon grain de sel, je vote pour le style :

```suggestion
    pricing_query = (
        db.session.query(finance_models.Pricing.id)
        .join(educational_models.CollectiveBooking, finance_models.Pricing.collectiveBooking)
        .join( educational_models.CollectiveStock, educational_models.CollectiveBooking.collectiveStock)
        .filter(
            educational_models.CollectiveStock.collectiveOfferId == collective_offer.id,
            finance_models.Pricing.status.in_(
                (finance_models.PricingStatus.PROCESSED, finance_models.PricingStatus.INVOICED)
            ),
        )
    )
```

Ça fait autant de ligne mais évite de répéter deux fois pricing_query à chaque ligne, et montre bien un bloc.

Reply:
avec l'indentation ça se fait en vrai

Reply:
c'est horrible vous vous liguez contre moi !

(pas de soucis, si je suis le seul a préferer ma méthode, je prendraits la votre :) )